### PR TITLE
chore: upgrade devDependencies around docs site to latest

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: starts on #5070
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

I upgraded many website-related packages to latest, then ran `npm audit fix --force`. 

Includes changes from #5069. ~Leaving in draft until that PR is approved, after which I'll merge those changes into this branch and un-draft. I'm sending now as a reference.~ ✅ 

There are a few vulnerabilities from old packages left after this. See https://github.com/mochajs/mocha/issues/5070#issuecomment-1871418480.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -130,13 +129,14 @@
     "rollup-plugin-visualizer": "^5.6.0",
     "sinon": "^9.0.3",
     "strip-ansi": "^6.0.0",
-    "svgo": "^1.3.2",
+    "svgo": "^3.1.0",
+    "taffydb": "^2.7.3",
```
Comment (JoshuaKGoldberg) [inline package.json:133]: This is required by `@mocha/docdash`. Once #5072 is resolved, we shouldn't need this explicit dep anymore.

**Comment by voxpelli** [comment]
This one needs to be updated now, but preferably it would wait until #5042 is merged

**Comment by voxpelli** [comment]
#5042 is now merged :)

**Comment by JoshuaKGoldberg** [comment]
Per #5207, we're going to go with a website overhaul instead.

**Pull request closed without merging**